File: 30 days of code/Day 14 - Scope/Scope.cs
/**
 *              SCOPE | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 14
 */

using System;
using System.IO;
using System.Linq;

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        int n = int.Parse(read.ReadLine());
        int[] array = read.ReadLine().Split(' ').Select(x => Convert.ToInt32(x)).ToArray();
        read.Close();

        Difference difference = new Difference(array);
        difference.ComputeDifference();
        Console.WriteLine(difference.maximumDifference);
    }
}

class Difference
{
    private int[] Elements { get; set; }
    public int maximumDifference;

    public Difference(int[] elements)
    {
        Elements = elements;
    }

    public void ComputeDifference()
    {
        maximumDifference = int.MinValue;
        for (int i = 0; i < Elements.Length - 1; ++i)
        {
            for (int j = i + 1; j < Elements.Length; ++j)
            {
                int difference = Math.Abs(Elements[i] - Elements[j]);
                maximumDifference = (difference > maximumDifference) ? difference : maximumDifference;
            }
        }
    }
}

File: 30 days of code/Day 14 - Scope/Scope.cpp
/**
*              SCOPE | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 14
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

ifstream file("input.txt");
istream& f = file;

class Difference {
public:
	int maximumDifference;

	Difference(vector<int> elements) {
		this->elements = elements;
	}

	void computeDifference() {
		maximumDifference = INT32_MIN;
		for (int i = 0; i < elements.size() - 1; ++i) {
			for (int j = 0; j < elements.size(); ++j) {
				int difference = abs(elements[i] - elements[j]);
				maximumDifference = (difference > maximumDifference) ? difference : maximumDifference;
			}
		}
	}
private:
	vector<int> elements;
};

void main() {
	int n; f >> n;
	vector<int> array(n);
	for (int i = 0; i < array.size(); ++i) {
		f >> array[i];
	}

	Difference difference(array);
	difference.computeDifference();
	cout << difference.maximumDifference << endl;
}